File: frontend/src/types/baselinePlan.ts
export type BaselinePlan = {
  id: number
  project_id: number
  name: string
  plan_type: string
  description?: string | null
  baseline_date?: string | null
  is_default: boolean
  is_active: boolean
  bound_batch_count: number
  latest_bound_batch_date?: string | null
  created_at: string
  updated_at: string
}

export type BaselinePlanPayload = Omit<BaselinePlan, 'id' | 'project_id' | 'bound_batch_count' | 'latest_bound_batch_date' | 'created_at' | 'updated_at'>

export type BaselineBoundBatch = {
  id: number
  project_id: number
  file_name: string
  sheet_name?: string | null
  data_date?: string | null
  status: string
  imported_count: number
  published_at?: string | null
  created_at: string
  baseline_plan_id?: number | null
  baseline_plan_name?: string | null
}

File: backend/app/schemas/baseline_plan.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class BaselinePlanBase(BaseModel):
    name: str
    plan_type: str = "current"
    description: str | None = None
    baseline_date: date | None = None
    is_default: bool = False
    is_active: bool = True


class BaselinePlanCreate(BaselinePlanBase):
    pass


class BaselinePlanUpdate(BaseModel):
    name: str | None = None
    plan_type: str | None = None
    description: str | None = None
    baseline_date: date | None = None
    is_default: bool | None = None
    is_active: bool | None = None


class BaselinePlanRead(BaselinePlanBase):
    id: int
    project_id: int
    bound_batch_count: int = 0
    latest_bound_batch_date: date | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class BaselineBoundBatch(BaseModel):
    id: int
    project_id: int
    file_name: str
    sheet_name: str | None = None
    data_date: date | None = None
    status: str
    imported_count: int = 0
    published_at: datetime | None = None
    created_at: datetime
    baseline_plan_id: int | None = None
    baseline_plan_name: str | None = None

    model_config = ConfigDict(from_attributes=True)
